core/src/main/java/org/apache/carbondata/core/metadata/schema/table/column/ColumnSchema.java

```diff
@@ -342,6 +342,30 @@ public void setParentColumnTableRelations(
     return true;
   }
 
+  /**
+   * method to compare columnSchema,
+   * other parameters along with just column name and column data type
+   * @param obj
+   * @return
+   */
+  public boolean equalsWithStrictCheck(Object obj) {
+    if (!this.equals(obj)) {
+      return false;
+    }
+    ColumnSchema other = (ColumnSchema) obj;
+    if (!columnUniqueId.equals(other.columnUniqueId) ||
+        (isDimensionColumn != other.isDimensionColumn) ||
+        (scale != other.scale) ||
+        (precision != other.precision) ||
+        (isSortColumn != other.isSortColumn)) {
+      return false;
+    }
+    if (encodingList.size() != other.encodingList.size()) {
```

Comment:
Better to check the encoding values also...this is a generic method and can be useful for other schenario

Reply:
done. added